### PR TITLE
Ports- Cloudy

### PR DESCRIPTION
**Pull request by CloudyLopez**
# Task List
Congratulations! You're submitting your assignment!
***pssssh- i tried
## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | model is where the logic belongs. [code]
Describe in your own words what the Controller is doing in Rails | controller is where all te 'actions' live
Describe in your own words what the View is doing in Rails | view is where the output is showing up
Describe an edge-case controller test you wrote | an edge case i wrote in destroy where you can try to find an invalid (wrong) id #.
What is the purpose of using strong params? (i.e. the params method in the controller) | strong params helps everything along. the other way i tried doing it didnt work so... strong params it is!
How are Rails migrations related to Rails models? | the rails migration corresponds to the models bc the migration does the structure.
Describe one area of Rails that are still unclear on | how to change one thing w/o fudging up everything else, why we need what when, how this works on a linear level, and SO MANY MORE.

### CODE REVIEW

**Comment by tildeee** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | x
Answered comprehension questions | x
Successfully handles: Index, Show | x
Index & Show tests pass | index tests pass
Successfully handles: New, Create | x
New & Create tests pass | 
Successfully handles: Edit, Update | x
Tests for Edit & Update test valid & invalid task ids | 
Successfully handles: Destroy, Task Complete | destroy works
Tests for Destroy & Task Complete include tests for valid and invalid task ids |
Routes follow RESTful conventions | x
Uses named routes (like `_path`) | x
**Overall** |



Hi Cloudy! Good work on this project!


You did a great job with doing the best practices for the code that you wrote. I'm really happy that you got all of the CRUD actions with the right RESTful routing and our best practices, with strong params, and proper redirecting.


There one thing that was missing that I'd like to bring attention to:
- Controller tests! In general, they were broken... because a `yml` file existed. However, if that file gets deleted, the tests run... and then mostly fail. My biggest concern for this is: I want to ensure that you can know and understand the theory and pattern of why and how we would write controller tests


It's OK to feel a little overwhelmed at this point, Rails is a lot! Make sure you're practicing vocabulary and reviewing how the theory of the request/response cycle and the MVC pattern connect to the actual code you're writing.

File: app/controllers/todos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  def new
+    @todo = Todo.new
+  end
+
+  def create
+    @todo = Todo.new(
+      name: params['todo']['name'],
+      completion_date: params['todo']['completion_date'],
+      description: params['todo']['description']
+    )
```
Comment (tildeee) [inline app/controllers/todos_controller.rb:23]: you can now update this to `@todo = Todo.new( todo_params )`
